core/internal/stream/sender.go

```diff
@@ -547,41 +497,52 @@ func (s *Sender) sendRequest(record *spb.Record, request *spb.Request) {
 // updateSettings updates the settings from the run record upon a run start
 // with the information from the server
 func (s *Sender) updateSettings() {
-	if s.startState == nil {
-		s.logger.CaptureError(
-			errors.New("sender: updateSettings: no start state"))
+	metadata, _ := s.streamRun.Metadata()
+	if s.settings == nil || metadata == nil {
 		return
 	}
 
 	// StartTime should be generally thought of as the Run last modified time
 	// as it gets updated at a run branching point, such as resume, fork, or rewind
-	if s.settings.GetStartTime().IsZero() && !s.startState.StartTime.IsZero() {
-		s.settings.UpdateStartTime(s.startState.StartTime)
+	startTime := metadata.StartTime()
+	if s.settings.GetStartTime().IsZero() && !startTime.IsZero() {
+		s.settings.UpdateStartTime(startTime)
 	}
 
+	runPath := metadata.RunPath()
+
 	// TODO: verify that this is the correct update logic
-	if s.startState.Entity != "" {
-		s.settings.UpdateEntity(s.startState.Entity)
+	if runPath.Entity != "" {
+		s.settings.UpdateEntity(runPath.Entity)
 	}
-	if s.startState.Project != "" {
-		s.settings.UpdateProject(s.startState.Project)
+	if runPath.Project != "" {
+		s.settings.UpdateProject(runPath.Project)
 	}
-	if s.startState.DisplayName != "" {
-		s.settings.UpdateDisplayName(s.startState.DisplayName)
+	if displayName := metadata.DisplayName(); displayName != "" {
+		s.settings.UpdateDisplayName(displayName)
 	}
 }
 
 // sendRequestRunStart sends a run start request to start all the stream
 // components that need to be started and to update the settings
 func (s *Sender) sendRequestRunStart(_ *spb.RunStartRequest) {
+	metadata, err := s.streamRun.Metadata()
+	if err != nil {
+		s.logger.CaptureError(
+			fmt.Errorf("sender: sendRequestRunStart: %v", err))
+		return
```

Comment:
should we return the error, otherwise we just wait the timeout

Reply:
I think there's never a response for RunStart, but good catch, let me check the other methods

Reply:
I checked all the methods. These ones don't need any changes because they never produce a response:

* sendTelemetry
* sendConfig
* sendSystemMetrics
* sendOutput
* sendMetric

The only one that needed a fix was sendRequestStopStatus, which I updated.